From 2cd55a32ca010df245c7fd7bd8b169d04b454b25 Mon Sep 17 00:00:00 2001
From: Larry Wu <larry.wu@intel.com>
Date: Tue, 29 Mar 2016 10:44:44 -0700
Subject: [PATCH 375/639] Title: Fix the ioremap error Issue: HSD 243893
 IssueType: Defect Symptom: ioremap error message during the boot. RootCause:
 The ioremap is hardcoded as non-cached. However, the dmi_base(0x3dfde000) is
 supposed to be cached memory. PAT memtype list: write-back @
 0x3df80000-0x3e020000 That is why we get this error message. ioremap error
 for 0x3dfde000-0x3dfdf000, requested 0x10 (_PAGE_CACHE_UC_MINUS), got 0x0
 (_PAGE_CACHE_WB) Resolution: Use ioremap_cache for dmi table Approvers:
 CCID-243893

---
 drivers/firmware/dmi_scan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/firmware/dmi_scan.c b/drivers/firmware/dmi_scan.c
index fa0affb..0f4395f 100644
--- a/drivers/firmware/dmi_scan.c
+++ b/drivers/firmware/dmi_scan.c
@@ -787,7 +787,7 @@ int dmi_walk(void (*decode)(const struct dmi_header *, void *),
 	if (!dmi_available)
 		return -1;
 
-	buf = ioremap(dmi_base, dmi_len);
+	buf = ioremap_cache(dmi_base, dmi_len);
 	if (buf == NULL)
 		return -1;
 
-- 
2.10.1

